src/Grpc.Core.Api/Internal/CodeAnalysisAttributes.cs

```diff
@@ -0,0 +1,151 @@
+#region Copyright notice and license
+
+// Copyright 2015-2016 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+// Content of this file is copied with permission from:
+// https://github.com/dotnet/runtime/tree/e2e43f44f1032780fa7c2bb965153c9da615c3d0/src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis
+// These types are intented to be added as internalized source to libraries and apps that want to
+// use them without requiring a dependency on .NET 5.
```

Comment:
There might already be an internal version of this file in the repo. Could bring it in via a link and remove this

Reply:
I wasn't able to find preexisting copy inside the grpc-dotnet repo.